expr/interval_year_to_month.go

```diff
@@ -0,0 +1,92 @@
+package expr
+
+import (
+	"fmt"
+
+	"github.com/substrait-io/substrait-go/proto"
+	"github.com/substrait-io/substrait-go/types"
+)
+
+// IntervalYearToMonthLiteral implements Literal interface for interval year to month type
+type IntervalYearToMonthLiteral struct {
+	years       int32
+	months      int32
+	nullability types.Nullability
+}
+
+func (m IntervalYearToMonthLiteral) WithYear(years int32) IntervalYearToMonthLiteral {
+	return IntervalYearToMonthLiteral{
+		years:       years,
+		months:      m.months,
+		nullability: m.nullability,
+	}
+}
```

Comment:
same comment as with IntervalCompoundLiteral

Reply:
fixed it